content/400-reference/200-api-reference/050-prisma-client-reference.mdx

```diff
@@ -552,23 +559,42 @@ const result = await prisma.user.findUnique({
 })
 ```
 
+### <inlinecode>findUniqueOrThrow</inlinecode>
+
+<Admonition type="info">
+
+We introduced `findUniqueOrThrow` in v4.0.0. It replaces the [`rejectOnNotFound`](#inlinecoderejectonnotfoundinlinecode) option. `rejectOnNotFound` is deprecated in v4.0.0.
+
+</Admonition>
+
+`findUniqueOrThrow` retrieves a single data record in the same way as [`findUnique`](#inlinecodefinduniqueinlinecode). However, if the query does not find a record, it returns `NotFoundError: No User found error`.
+
+`findUniqueOrThrow` differs from `findUnique` as follows:
+
+- Its return type is non-nullable. For example, `post.findUnique()` can return `post` or `null`, but `post.findUniqueOrThrow()` always returns `post`.
+- It is not compatible with the [`$transaction` API](/concepts/components/prisma-client/transactions#the-transaction-api).
```

Comment:
It is only incompatible with "array of promises" style API, interactive transactions work as expected. In fact, I'd suggest interactive transactions as a workaround. 

Maybe also worth mentioning in what way exactly it is incompatible: `NotFoundError`, thrown by these methods won't rollback the transaction. If query fails for any other reason on the database side, transaction will roll back normally.